### PR TITLE
backport: merge bitcoin#16337, #16378, #20179, #19969 (send rpc)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

Extracted from [dash#5842](https://github.com/dashpay/dash/pull/5842) due to breaking changes and placed into its own PR to be merged during new major version development cycle.

## Breaking Changes

_(Taken from `release-notes-5861.md`)_

- The `walletcreatefundedpsbt` RPC call will now fail with `Insufficient funds` when inputs are manually selected but are not enough to cover the outputs and fee. Additional inputs can automatically be added through the new `add_inputs` option.

- The `fundrawtransaction` RPC now supports `add_inputs` option that when `false` prevents adding more inputs if necessary and consequently the RPC fails.

- A new `send` RPC with similar syntax to `walletcreatefundedpsbt`, including support for coin selection and a custom fee rate. The `send` RPC is experimental and may change in subsequent releases. Using it is encouraged once it's no longer experimental: `sendmany` and `sendtoaddress` may be deprecated in a future release.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by PastaPastaPasta — APPROVED** [review]
utACK c1fd5a622317b2d8bce4f2984970348e723419b1 for merging via merge commit

**Review by UdjinM6 — COMMENTED** [review]
few small suggestions

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3968,6 +3982,176 @@ static UniValue listlabels(const JSONRPCRequest& request)
...
+            RPCResult::Type::OBJ, "", "",
+                {
+                    {RPCResult::Type::BOOL, "complete", "If the transaction has a complete set of signatures"},
+                    {RPCResult::Type::STR_HEX, "txid", "The transaction id for the send. Only 1 transaction is created regardless of the number of addresses."},
+                    {RPCResult::Type::STR_HEX, "hex", "If add_to_wallet is false, the hex-encoded raw transaction with signature(s)"},
+                    {RPCResult::Type::STR, "psbt", "If more signatures are needed, or if add_to_wallet is false, the base64-encoded (partially) signed transaction"}
+                }
+        },
+        RPCExamples{""
+        "\nSend with a fee rate of 1 satoshi per byte\n"
```
Comment (knst) [inline src/wallet/rpcwallet.cpp:4052]: 1duff per byte?

**Review by knst — APPROVED** [review]
LGTM but check rpc dashification s/satoshi/duffs

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3968,6 +3982,176 @@ static UniValue listlabels(const JSONRPCRequest& request)
...
+    RPCHelpMan{"send",
+        "\nSend a transaction.\n",
+        {
+            {"outputs", RPCArg::Type::ARR, RPCArg::Optional::NO, "a json array with outputs (key-value pairs), where none of the keys are duplicated.\n"
+                    "That is, each address can only appear once and there can only be one 'data' object.\n"
+                    "For convenience, a dictionary, which holds the key-value pairs directly, is also accepted.",
+                {
+                    {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                        {
+                            {"address", RPCArg::Type::AMOUNT, RPCArg::Optional::NO, "A key-value pair. The key (string) is the dash address, the value (float or string) is the amount in " + CURRENCY_UNIT + ""},
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/wallet/rpcwallet.cpp]: Resolved in latest push

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (kwvg) [reply src/wallet/rpcwallet.cpp]: Resolved in latest push

**Review by knst — DISMISSED** [review]
utACK

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -3960,6 +3982,176 @@ static UniValue listlabels(const JSONRPCRequest& request)
...
+                {
+                    {RPCResult::Type::BOOL, "complete", "If the transaction has a complete set of signatures"},
+                    {RPCResult::Type::STR_HEX, "txid", "The transaction id for the send. Only 1 transaction is created regardless of the number of addresses."},
+                    {RPCResult::Type::STR_HEX, "hex", "If add_to_wallet is false, the hex-encoded raw transaction with signature(s)"},
+                    {RPCResult::Type::STR, "psbt", "If more signatures are needed, or if add_to_wallet is false, the base64-encoded (partially) signed transaction"}
+                }
+        },
+        RPCExamples{""
+        "\nSend with a fee rate of 1 duff per byte\n"
+        + HelpExampleCli("send", "'{\"" + EXAMPLE_ADDRESS[0] + "\": 0.1}' 1 sat/b\n" +
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:4053]:
```suggestion
        + HelpExampleCli("send", "'{\"" + EXAMPLE_ADDRESS[0] + "\": 0.1}' 1 duff/b\n" +
```

**Review by UdjinM6 — COMMENTED** [review]
noticed another one

File: src/wallet/rpcwallet.cpp [hunk 4]
```diff
@@ -3960,6 +3982,176 @@ static UniValue listlabels(const JSONRPCRequest& request)
...
+            RPCResult::Type::OBJ, "", "",
+                {
+                    {RPCResult::Type::BOOL, "complete", "If the transaction has a complete set of signatures"},
+                    {RPCResult::Type::STR_HEX, "txid", "The transaction id for the send. Only 1 transaction is created regardless of the number of addresses."},
+                    {RPCResult::Type::STR_HEX, "hex", "If add_to_wallet is false, the hex-encoded raw transaction with signature(s)"},
+                    {RPCResult::Type::STR, "psbt", "If more signatures are needed, or if add_to_wallet is false, the base64-encoded (partially) signed transaction"}
+                }
+        },
+        RPCExamples{""
+        "\nSend with a fee rate of 1 duff per byte\n"
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/wallet/rpcwallet.cpp]: Good catch!

File: src/wallet/rpcwallet.cpp (hunk 3 above)
Comment (kwvg) [reply src/wallet/rpcwallet.cpp]: Used `CURRENCY_ATOM`, resolved in latest push.

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK 76a49ad

**Review by knst — APPROVED** [review]
utACK 76a49addd06a69a463c68621bcc189c2a4ecac3e

**Review by PastaPastaPasta — APPROVED** [review]
utACK 76a49addd06a69a463c68621bcc189c2a4ecac3e

**Pull request merged**